### PR TITLE
Updating samples to use xunit

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes:
- https://github.com/dotnet/roslyn/issues/6843
- https://github.com/dotnet/roslyn/issues/6422
- https://github.com/dotnet/roslyn/issues/5918

### CODE REVIEW

**Comment by jmarolf** [comment]
@dotnet/roslyn-ide

File: src/Samples/CSharp/APISampleUnitTests/FAQ.cs [hunk 1]
```diff
@@ -888,15 +886,15 @@ public static void Main()
             SyntaxToken tokenNotFromTree = SyntaxFactory.Token(SyntaxKind.ClassKeyword);
             SyntaxNode nodeNotFromTree = other.GetRoot();
 
-            Assert.IsTrue(nodeFromTree.SyntaxTree == tree);
-            Assert.IsTrue(nodeFromTree.SyntaxTree == model.SyntaxTree);
-            Assert.IsFalse(tokenNotFromTree.SyntaxTree == tree);
-            Assert.IsFalse(nodeNotFromTree.SyntaxTree == model.SyntaxTree);
-            Assert.IsTrue(nodeNotFromTree.SyntaxTree == other);
+            Assert.True(nodeFromTree.SyntaxTree == tree);
```
Comment (jasonmalinowski) [inline src/Samples/CSharp/APISampleUnitTests/FAQ.cs]: Assert.Equal?

File: src/Samples/CSharp/APISampleUnitTests/FAQ.cs [hunk 2]
```diff
@@ -1651,7 +1649,7 @@ static void Method3()
                                                        syntaxTrees: new[] { tree },
                                                        references: new[] { Mscorlib });
             var diagnostics = compilation.GetDiagnostics();
-            Assert.AreEqual(0, diagnostics.Count());
+            Assert.Equal(0, diagnostics.Count());
```
Comment (jasonmalinowski) [inline src/Samples/CSharp/APISampleUnitTests/FAQ.cs]: I think xUnit has Assert.Empty?

File: src/Samples/Samples.sln [hunk 3]
```diff
@@ -17,8 +17,6 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "ImplementNotifyPropertyChan
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MakeConstCS", "CSharp\MakeConst\Impl\MakeConstCS.csproj", "{DE9D901F-43D6-4559-A068-B6E28E04F12E}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "MakeConstCS.UnitTests", "CSharp\MakeConst\Test\MakeConstCS.UnitTests.csproj", "{BFD0109A-767E-41F6-8EF3-1A75FB4F5E8D}"
```
Comment (brettfo) [inline src/Samples/Samples.sln]: MakeConstCS exists but no tests?

File: src/Samples/VisualBasic/APISampleUnitTests/FAQ.vb [hunk 4]
```diff
@@ -1206,14 +1201,14 @@ End Module
             ' Get TypeSymbol corresponding to above ModifiedIdentifierSyntax.
             Dim variableType = (CType(model.GetDeclaredSymbol(modifiedIdentifier), ILocalSymbol)).Type
 
-            Assert.IsFalse(variableType.Equals(structType)) ' Type of variable is a closed generic type while that of the struct is an open generic type.
-            Assert.IsTrue(variableType.OriginalDefinition.Equals(structType)) ' OriginalDefinition for a closed generic type points to corresponding open generic type.
-            Assert.AreEqual("NS.C(Of Integer).S(Of Long)", variableType.ToDisplayString())
-            Assert.AreEqual("Global.NS.C(Of Integer).S(Of Long)", variableType.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat))
+            Assert.False(variableType.Equals(structType)) ' Type of variable is a closed generic type while that of the struct is an open generic type.
```
Comment (jasonmalinowski) [inline src/Samples/VisualBasic/APISampleUnitTests/FAQ.vb]: Assert.NotEqual

**Comment by jasonmalinowski** [comment]
:+1:. Perhaps the commit should be broken up between "move to xunit" and "delete other broken stuff"?

**Comment by brettfo** [comment]
:shipit:

File: src/Samples/Samples.sln (hunk 3 above)
Comment (jmarolf) [inline src/Samples/Samples.sln]: MakeConstCS tests have not been run in... *One Thousand Years*.  If they provide any benefit, we've somehow learned to live without it

**Comment by jmarolf** [comment]
@dotnet-bot retest prtest/lin/dbg/unit32 please
// Previous failure: http://dotnet-ci.cloudapp.net/job/roslyn_prtest_lin_dbg_unit32/2579/
// Retest reason:

**Pull request merged**